test/Main.hs

```diff
@@ -49,18 +50,12 @@ parseTemplate
       , "{}"
       ]
 
-    mkAccepted :: (Text, Paths.Template) -> TestTree
-    mkAccepted (input, expected)
-      = testCase (convertText ("accepts: " <> show input))
-      $ pure expected @=? parseJSON input
-
-    mkRejected :: Text -> TestTree
-    mkRejected input
-      = testCase (convertText ("rejects: " <> show input))
-      $ Left ("Error in $: invalid template path: " <> show input) @=? parseJSON input
-
-    parseJSON :: Text -> Either String Paths.Template
-    parseJSON = JSON.parseEither JSON.parseJSON . JSON.String
+    mkRejected' :: Text -> TestTree
+    mkRejected' input =
+      mkRejected
+        (JSON.parseJSON @Paths.Template)
+        "Error in $: invalid template path:"
```

Comment:
@mbj Should there be a trailing space after the final `:`?

Reply:
no, ids added by the downstream code.